lib/ansible/modules/cloud/amazon/cloudformation.py

```diff
@@ -509,6 +506,18 @@ def main():
     except botocore.exceptions.NoCredentialsError as e:
         module.fail_json(msg=boto_exception(e))
 
+    # Wrap the cloudformation client methods that this module uses with
+    # automatic backoff / retry for throttling error codes
+    backoff_wrapper = AWSRetry.jittered_backoff(retries=10, delay=3, max_delay=30)
+    cfn.describe_stack_events = backoff_wrapper(cfn.describe_stack_events)
+    cfn.create_stack = backoff_wrapper(cfn.create_stack)
```

Comment:
I'm not a huge fan either but the alternative looks pretty nasty as well and requires a significant amount of refactoring. Any method that currently has a `try: catch Exception:` block will be "immune" to retry decorators as the throttling exception never gets the chance to bubble up.

We end up needing to be very explicit in how we write functions or end up creating wrapped functions for each call that we want to make retry-able.

```python
# We can't add retries to methods like this
def some_method():
    try:
        cfn.describe_stack_events(args)
    catch Exception as e:
        module_fail(...)
``` 

```python
# we could do something like this
@AWSRetry.jittered_backoff(retries=10, delay=3, max_delay=30)
def describe_stacks(stack_name):
    return cfn.describe_stacks(StackName=stack_name)
```

```python
# potential solution that doesn't monkey patch anything and lets us retry requests
backoff_wrapper = AWSRetry.jittered_backoff(retries=10, delay=3, max_delay=30)
describe_stacks = backoff_wrapper(cfn.describe_stacks)

def some_method():
    try:
        describe_stacks(StackName=stack_name)
    catch Exception as e:
        module_fail(...)
```